jdaviz/configs/default/plugins/line_lists/line_lists.py

```diff
@@ -88,29 +106,173 @@ def _on_viewer_data_changed(self, msg=None):
             return
 
+        label = msg.data.label
         try:
-            viewer_data = self.app.get_viewer('spectrum-viewer').data()
+            viewer_data = self.app.get_data_from_viewer('spectrum-viewer').get(label)
         except TypeError:
             warn_message = SnackbarMessage("Line list plugin could not retrieve data from viewer",
                                            sender=self, color="error")
             self.hub.broadcast(warn_message)
             return
 
         # If no data is currently plotted, don't attempt to update
-        if viewer_data is None or len(viewer_data) == 0:
+        if viewer_data is None:
+            return
+
+        self._units["x"] = str(viewer_data.spectral_axis.unit)
+        self._units["y"] = str(viewer_data.flux.unit)
+
+        self._bounds["min"] = viewer_data.spectral_axis[0]
+        self._bounds["max"] = viewer_data.spectral_axis[-1]
+
+        # set redshift slider to redshift stored in Spectrum1D object
+        self.rs_redshift = viewer_data.redshift.value
+        self._auto_slider_range()  # will also trigger _auto_slider_step
+
+    def _parse_redshift_msg(self, msg):
+        '''
+        Handle incoming redshift messages from the app hub. Generally these
+        will be created by Specviz helper methods.
+        '''
+        if msg.sender == self:
             return
 
-        self._viewer_spectrum = viewer_data[0]
+        param = msg.param
+
+        if param == "rs_slider_range":
+            if msg.value == 'auto':
+                # observer will handle setting rs_slider_range
+                self.rs_slider_range_auto = True
+            else:
+                self.rs_slider_range_auto = False
+                self.rs_slider_half_range = float(msg.value)/2
+        elif param == "rs_slider_step":
+            if msg.value == 'auto':
+                # observer will handle setting rs_slider_step
+                self.rs_slider_step_auto = True
+            else:
+                self.rs_slider_step_auto = False
+                self.rs_slider_step = float(msg.value)
+        elif param == "redshift":
+            # NOTE: this should trigger the observe to update rs_rv, line positions, and
+            # update self._global_redshift
+            self.rs_redshift = float(msg.value)
+        elif param == 'rv':
+            # NOTE: this should trigger the observe to update rs_redshift, line positions, and
+            # update self._global_redshift
+            self.rs_rv = float(msg.value)
+        else:
+            raise NotImplementedError(f"RedshiftMessage with param {param} not implemented.")
 
-        self._units["x"] = str(self._viewer_spectrum.spectral_axis.unit)
-        self._units["y"] = str(self._viewer_spectrum.flux.unit)
+    def _velocity_to_redshift(self, velocity):
+        """
+        Convert a velocity to a relativistic redshift.
+        """
+        beta = velocity / c
```

Comment:
```suggestion
        beta = velocity / const.c
```

Also, this is exactly the same as https://github.com/astropy/astropy/blob/5b6888e17f9c9536af7c554f91d0e04929f963d6/astropy/coordinates/spectral_coordinate.py#L37 though I am torn between re-implementation and using a hidden function in `astropy`. Maybe we have to wait for astropy/astropy#10240 ? 😞